### PR TITLE
br/restore: try keep table ID when possible

### LINKED ISSUE #38438: br/restore: try to keep table ID when possible for speed up restoration
## Enhancement
Just check the PDF document: 
[Restoration Optimization: Eliminating Key Rewriting.pdf](https://github.com/pingcap/tidb/files/9769971/Restoration.Optimization_.Eliminating.Key.Rewriting.pdf)

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

Issue Number: close #38438

Problem Summary:
Key Rewriting is a heavy operation and may slow down the restoration. We can try to reduce it when possible.

### What is changed and how it works?
This PR make BR try to keep the table ID when restoring.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

|Field| With reuse table ID | Without reuse table ID |
| - | - | - |
|CPU Usage|image|image|
|Speed|image|image|

- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Speed up restoration of new clusters.
```

### CODE REVIEW

**Review by 3pointer — APPROVED** [review]

File: br/pkg/utils/key.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ import (
 	"go.uber.org/zap"
 )
 
+type TableIDs struct{}
```
Comment (Leavrth) [inline br/pkg/utils/key.go:20]: is it still needed?

**Review by Leavrth — APPROVED** [review]

File: br/pkg/utils/key.go (hunk 1 above)
Comment (YuJuncen) [reply br/pkg/utils/key.go:20]: I guess it can be removed🤔

**Pull request merged**